caffe2/contrib/aten/aten_op_template.h

```diff
@@ -89,16 +89,18 @@ class ATenOp : public Operator<Context> {
     CAFFE_THROW("Unknown type meta"); // TODO: improve error message...
   }
   void assignTo(Tensor* dst, const at::Tensor& src_) {
-    at::Tensor src = src_.contiguous();
-    auto at_sizes = src.sizes();
+    at::Tensor* src = new at::Tensor(src_.contiguous());
+    auto at_sizes = src->sizes();
     std::vector<int64_t> dims(at_sizes.begin(),at_sizes.end());
     dst->Resize(dims);
     dst->ShareExternalPointer(
-        src.data_ptr(), typeMetaFor(src), 0, [src](void* ptr) mutable {
-          // return a closure that holds a handle to t until it is called
-          // to keep the aten memory alive
-          return src.reset();
-        });
+        at::DataPtr(
+            (void*)src->data_ptr(),
+            static_cast<void*>(src),
+            [](void* t_ptr) -> void { delete static_cast<at::Tensor*>(t_ptr); },
+            src->device()),
+        typeMetaFor(*src),
+        0);
```

Comment:
You... didn't make things any worse, but I find it hard to believe this ever worked with `capacity = 0`

Reply:
I think this function might actually be unused aside from assignListStartingAt which is also unused - checking right now.